x/poolmanager/router_test.go

```diff
@@ -1940,8 +2126,91 @@ func (s *KeeperTestSuite) TestSplitRouteExactAmountIn() {
 
 			// 1x from single route two hops and 3x from single route three hops
 			expectedTokenOutEstimate: twentyFiveBaseUnitsAmount.MulRaw(4),
-		},
+			expectedTakerFees:        zeroTakerFeeDistr,
+		},
+		"valid split route multi hop (exact output check)": {
+			// Set the pools we swap through to all have a 0.35% taker fee
+			setupPools: s.withTakerFees(
+				defaultValidPools,
+				[]uint64{0, 3, 6, 7},
+				[]sdk.Dec{pointThreeFivePercent, pointThreeFivePercent, pointThreeFivePercent, pointThreeFivePercent},
+			),
+			routes: []types.SwapAmountInSplitRoute{
+				withInputSwapIn(defaultSingleRouteTwoHops, sdk.NewInt(1000)),
+				withInputSwapIn(fooAbcBazTwoHops, sdk.NewInt(1000)),
+			},
+			tokenInDenom:      foo,
+			tokenOutMinAmount: sdk.OneInt(),
 
+			// We charge taker fee on each hop and expect the output to truncate
+			// The output of first hop: (1 - 0.0035) * 1000 = 996.5, truncated to 996 (4foo taker fee)
+			// which has an additional truncation after the swap executes, leaving 995.
+			// The second hop is similar: (1 - 0.0035) * 995 = 991.5, truncated to 991 (4bar taker fee)
+			// which has an additional truncation after the swap executes, leaving 990.
+			expectedTokenOutEstimate: sdk.NewInt(990 + 990),
+			checkExactOutput:         true,
+
+			// Expected taker fees from calculation above are:
+			// * [4foo, 4bar] for first route
+			// * [4foo, 4abc] for second route
+			// Due to truncation, 4 units of fees get distributed 1 to community pool, 3 to stakers.
+			// Recall that foo and bar are quote assets, while abc is not.
```

Comment:
Happy to shorten comments if this is too much – mostly wanted to make sure rounding behavior was properly dug into here

Reply:
No, this is appreciated